File: src/main/java/io/curiousoft/ijudi/ordermanagement/model/Bank.java
package io.curiousoft.ijudi.ordermanagement.model;

import javax.validation.constraints.NotBlank;

public class Bank {

    @NotBlank(message = "Bank name not valid")
    private String name;
    @NotBlank(message = "Bank phone not valid")
    private String phone;
    @NotBlank(message = "Bank account id not valid")
    private String accountId;
    private String type;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAccountId() {
        return accountId;
    }

    public void setAccountId(String accountId) {
        this.accountId = accountId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/model/Bank.kt
package io.curiousoft.izinga.commons.model

import javax.validation.constraints.NotBlank

class Bank {
    var branchCode: @NotBlank(message = "Bank Branch code not valid") String? = null
    var name: @NotBlank(message = "Bank name not valid") String? = null
    var phone: @NotBlank(message = "Bank phone not valid") String? = null
    var accountId: @NotBlank(message = "Bank account id not valid") String? = null
    var type: BankAccType? = null
}

enum class BankAccType(val code: String) {
    CHEQUE("1"), EWALLET("D"), SAVINGS("2"), TRANSMISSION("3"), wallet(""), string("")
}